src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,61 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final String BASIC_DELIMITER = ",|:";
+    private static final int EMPTY_NULL_RETURN_VALUE = 0;
+
+
+    public static int calculateSum(String strValue) {
+        if (isEmptyOrNull(strValue))
+            return EMPTY_NULL_RETURN_VALUE;
+
+        return split(StringsToInts(getStringValues(strValue, getMatcher(strValue))));
+    }
+
+    private static String getDelimiter(Matcher m) {
+        if (m.find()) {
+            return BASIC_DELIMITER + "|" + m.group(1);
+        }
+        return BASIC_DELIMITER;
+    }
+
+    private static String[] getStringValues(String strValue, Matcher m) {
+        String delimiter = getDelimiter(getMatcher(strValue));
+        if (m.find()) {
+            return m.group(2).split(delimiter);
+        }
+        return strValue.split(delimiter);
+    }
+
+    private static Matcher getMatcher(String strValue) {
+        return Pattern.compile("//(.)\n(.*)").matcher(strValue);
+    }
+
+    private static int split(int[] numbers) {
+        int result = 0;
+        for (int number : numbers) {
+            result += number;
+        }
+        return result;
+    }
```

Comment:
```suggestion
    private static int sum(int[] numbers) {
        int result = 0;
        for (int number : numbers) {
            result += number;
        }
        return result;
    }
```

메소드의 행위와 이름을 봤을 때 매칭되지 않네요. sum 같은 이름이 적절할 것 같습니다 😄